Dessert Recipes Project  – Technigo Bootcamp Project 2025

KausarShangareeva opened the pull request:
Hi there! 👋
I built Recipe Library as part of my **Technigo JavaScript Bootcamp 2025 journey.** This project lets users explore and save dessert recipes from different cuisines in a fun and interactive way.

> Check out the live website here: [Recipe Library](https://favorate-drink.netlify.app/)

Key Features:

- Filter desserts by cuisine (Italy / Asia / Middle East)
- Random dessert generator for fun inspiration
- Favorites saved locally using LocalStorage
- Handles API limits and caches data for offline use
- Fully responsive design for mobile, tablet, and desktop

This project helped me practice working with APIs, dynamic content, local storage, and creating interactive, responsive UI components. 🌟

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
Push latest changes

KausarShangareeva committed:
chore: push latest changes

KausarShangareeva committed:
chore: push latest changes

KausarShangareeva committed:
chore: push latest changes

KausarShangareeva committed:
chore: push latest changes

irisdgz commented on index.html:
I think your HTML structure is nice and clean! It’s easy to see where each part of the app goes. Super nice you used main and section for the recipes whitch that makes it very readable. Maybe you could add an alt attribute (like a short description) to your images so it helps people using screen readers know what the picture shows :)

irisdgz commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+const LS_COFFEE = "coffeeData";
+const LS_JUICE = "juiceData";
+const LS_FAVORITES = "favorites";
+
```
Good use of localStorage keys!

KausarShangareeva replied:
Thank!)

irisdgz commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+  buttons.filter(Boolean).forEach((btn) => btn.classList.remove("active"));
+  if (activeBtn) activeBtn.classList.add("active");
+}
+
```
I recognize this pattern from I dont remeber where, youtube or our session! Removing and adding classes to show active buttons. Nice and clean implementation!

irisdgz commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+  if (coffeeRendered) return;
+  coffeeRendered = true;
+  if (containerCoffee) containerCoffee.innerHTML = "";
+
```
Nice use of async/await to fetch the recipe data!

irisdgz commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+
+  ensureNoMatchBanner(containerCoffee);
+}
+
```
Nice how you check localStorage first before fetching new data , here learned something myself as well :)

irisdgz commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+  const isCoffee = category === "coffee";
+  const container = isCoffee ? containerCoffee : containerJuice;
+  if (!container) return;
+
```
well used async here again!

irisdgz commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+  if (!cards || !cards.length) return null;
+  return cards[Math.floor(Math.random() * cards.length)];
+}
+
```
Good job here with random recpies.

irisdgz commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+    item.parentElement
+      ?.querySelectorAll(".filter__item")
+      .forEach((li) => li.classList.remove("active"));
+    item.classList.add("active");
```
This looks great! I like how you combine event listeners with sorting and filtering.

irisdgz commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+    console.warn("No data-id on card");
+    return;
+  }
+
```
Cool how you added a favorites feature! Great way to use localStorage and JSON to save user interactions

irisdgz reviewed (commented):
Really nice work Kausyar! The code is well structured and realtivly easy for me to follow. I like how you’ve added clear sections for rendering, filters and favorites. Cool to see how you’re using async/await and localStorage together. The filtering and sorting part is especially nice and works smoothly. Overall, super solid project eventhou it has its own style and not the one in the Figma demo!

JennieDalgren commented on css/style.css:
```diff
@@ -0,0 +1,790 @@
+@charset "UTF-8";
+/* ------------------------------------------------------
+  | THEME VARIABLES: базовые (light) ☀️
+  ------------------------------------------------------ */
+:root {
+  --color-primary-first-light: #fafbff;
+  --color-primary-first: #0018a4;
+  --color-primary-second-light: #ffecea;
+  --color-primary-second: #ff6589;
+  --color-white: #ffffff;
+  --color-white-base: #ffffff;
+  --color-grey-light-1: #f8f9fa;
+  --color-grey-light-2: #e9ecef;
+  --color-grey-light-3: #dee2e6;
+  --color-grey-light-4: #ced4da;
+  --color-grey-dark: #868e96;
+  --color-grey-dark-base: #868e96;
+  --color-font: #212529;
+  --color-font-base: #212529;
+  --color-button: #8d9271;
+  --color-black: #000000;
+  --color-blue: #176d8c;
+  --border-radius-xxs: 0.5rem;
+  --border-radius-xs: 1rem;
+  --border-radius-sm: 1.5rem;
+  --border-radius-md: 2rem;
+  --border-radius-lg: 2.5rem;
+  --border-radius-xl: 3rem;
+  --box-shadow: 0 0.5rem 1rem #0019a453;
```
nice that you use variables

JennieDalgren commented on css/style.css:
```diff
@@ -0,0 +1,790 @@
+  ------------------------------------ */
+/* 
+
+ПЕРЕКЛЮЧЕНИЯ МЕЖДУ КАРТОЧКАМИ
```
stick to english comments

KausarShangareeva replied:
Im done ✅✅✅

JennieDalgren commented on css/style.css:
```diff
@@ -0,0 +1,790 @@
+}
+.header__input-mobile {
+  border: 1px solid #e9ecef;
+  background-color: #e9ecef;
```
use the variable that you have for this color?

KausarShangareeva replied:
Im done ✅✅✅

JennieDalgren commented on css/style.css:
```diff
@@ -0,0 +1,790 @@
+}
+.header__input-mobile:focus {
+  outline: none;
+  border-color: #e9ecef;
```
variable?

KausarShangareeva replied:
Im done ✅✅✅

JennieDalgren commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+   =============== */
+const filterItems = document.querySelectorAll(".filter__item");
+
+const mainCoffee = document.querySelector(".main-coffee");
+const mainJuice = document.querySelector(".main-juice");
+const mainTea = document.querySelector(".main-tea");
+const mainFavorites = document.querySelector(".main-favorites");
```
I'm very confused about all these drink references? That is not reflected in your deployed website, which looks like it is focused on desserts?

KausarShangareeva replied:
Im done ✅✅✅ I wrote desserts instead of juice

JennieDalgren commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+  const ing = ingredients.map((i) => String(i || "").toLowerCase()).join(" ");
+  if (/(mascarpone|savoiardi|amaretto)/.test(ing)) return "italian";
+  if (/(matcha|azuki|black\s*sesame|rice\s*flour)/.test(ing)) return "asian";
+  if (/(tahini|date|rose\s*water|orange\s*blossom|cardamom)/.test(ing))
```
This looks very hard coded, why don't you use the cuisine data from the API instead?

KausarShangareeva replied:
I simplified the cuisine detection function (detectCuisine). Now it only uses the cuisines field from the api data, and if the cuisine is not specified, it returns 'international'

JennieDalgren commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+   ⁉️ Converts any value to number safely (NaN → 0)
+   ================================== */
+
+function toNum(v) {
+  const n = Number(v);
+  return Number.isFinite(n) ? n : 0;
+}
```
what do you need this for?

KausarShangareeva replied:
The toNum function is used for safely converting values to numbers when sorting the cards 😀

JennieDalgren commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+}
+
+window.addEventListener("DOMContentLoaded", () => {
+  seedIfEmpty(LS_COFFEE, DEMO_CAKES);
+  seedIfEmpty(LS_JUICE, DEMO_DESSERTS);
+
```
should this actually be happening? You are supposed to fetch from the API?

KausarShangareeva replied:
You're totally right to ask! Here’s the idea behind seedIfEmpty:

1- Fallback for API limits:
The Spoonacular API has strict limits (150 requests/day on the free plan). When the limit is reached, it returns a 402/429 error. Demo data makes sure the filters always work so you can test them.

2- Immediate UI feedback:
Without demo data, the user would see a blank screen while waiting for the API. With demo data, the interface shows content right away and updates it if the API responds.

JennieDalgren commented on js/script.js:
```diff
@@ -0,0 +1,911 @@
+  mainJuice?.classList.remove("hidden");
+  mainTea?.classList.add("hidden");
+  mainFavorites?.classList.add("hidden");
+  setActiveButton(buttonJuice);
```
I get the logic here, and it is nice. However I don't see any use of a coffee or juice button? Use names that makes sense for your project.

KausarShangareeva replied:
done ✅

JennieDalgren reviewed (changes requested):
Good job with this project!

Take a look at the API calls and make sure yoou have data to be able to try your filters. I get empty states on almost all of them.

Really good implementation of favorite recipes and that is persists in local storage.

You project is not entirely responsive, see the screenshot:
<img width="1512" height="982" alt="image" src="https://github.com/user-attachments/assets/e1a53192-74b9-421d-811f-50aa60e0e7fa" />

Clean up your code a bit and work on the naming of your functions. If you change your "theme" of the recipes, change the names in your code too. Also, stick to english comments. And remember to use commit messages in a better way, the message should reflect the change that you have done in the code, do not just copy paste the same message every time, that really beats the purpose of commits.

What you need to fix:
- Responsiveness
- Make sure the filters you have chosen to work with is actually working
- Clean up the code. Remove unused code, proper naming of variables and functions, english in comments if you need them.

Almost there! 💪

KausarShangareeva committed:
Revise README for Favorite Dessert project

Updated project description and added features, demo, technologies, and file structure.

KausarShangareeva committed:
Simplify README introduction text

Removed line breaks for a more concise introduction.

KausarShangareeva committed:
Add

KausarShangareeva committed:
Update README with screenshot and features section

KausarShangareeva committed:
Add screenshot to README

KausarShangareeva commented:
> Good job with this project!
> 
> Take a look at the API calls and make sure yoou have data to be able to try your filters. I get empty states on almost all of them.
> 
> Really good implementation of favorite recipes and that is persists in local storage.
> 
> You project is not entirely responsive, see the screenshot: <img alt="image" width="1512" height="982" src="https://private-user-images.githubusercontent.com/18549785/505766426-e1a53192-74b9-421d-811f-50aa60e0e7fa.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NjMzMDA0NTAsIm5iZiI6MTc2MzMwMDE1MCwicGF0aCI6Ii8xODU0OTc4NS81MDU3NjY0MjYtZTFhNTMxOTItNzRiOS00MjFkLTgxMWYtNTBhYTYwZTBlN2ZhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTExMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMTE2VDEzMzU1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJlMzRhYzU4MGQwMGM4ZjRjNjQ2ZTZkOTgyZmMwN2JjYjAxOThkMTg2ODdhNzA5Yzc2Y2ZiNGMxY2ViYzM5Y2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.2z8_EO85qnq0fmRy-vdo8ZpMI4HkWFCf8gUU1AlNYxc">
> 
> Clean up your code a bit and work on the naming of your functions. If you change your "theme" of the recipes, change the names in your code too. Also, stick to english comments. And remember to use commit messages in a better way, the message should reflect the change that you have done in the code, do not just copy paste the same message every time, that really beats the purpose of commits.
> 
> What you need to fix:
> 
> * Responsiveness
> * Make sure the filters you have chosen to work with is actually working
> * Clean up the code. Remove unused code, proper naming of variables and functions, english in comments if you need them.
> 
> Almost there! 💪

Im done ✅✅✅

KausarShangareeva committed:
Update project files and README

KausarShangareeva committed:
Deleted not used img

KausarShangareeva committed:
Deleted not needed js

JennieDalgren reviewed (approved):
✅
